telemetry/validation/src/validation.ts

```diff
@@ -0,0 +1,47 @@
+import { stringSort } from './sorting'
+import { TelemetryDefinitions } from './telemetryDefinitions'
+
+export function validate(definitions: TelemetryDefinitions): string[] {
+    const validations: string[] = []
+
+    validations.push(...validateTypeOrder(definitions))
+    validations.push(...validateMetricsOrder(definitions))
+
+    return validations
+}
+
+function validateTypeOrder(definitions: TelemetryDefinitions): string[] {
+    const validations = []
+
+    const sortedNames = definitions.types.map(t => t.name).sort(stringSort)
+
+    for (let i = 0; i < definitions.types.length; i++) {
+        const inputType = definitions.types[i]
+        const sortedName = sortedNames[i]
+
+        if (inputType.name != sortedName) {
+            validations.push(`Telemetry Types are not sorted. Expected: ${sortedName}, Found: ${inputType.name}`)
+            break
+        }
+    }
+
+    return validations
+}
+
+function validateMetricsOrder(definitions: TelemetryDefinitions): string[] {
+    const validations = []
+
+    const sortedNames = definitions.metrics.map(t => t.name).sort(stringSort)
+
+    for (let i = 0; i < definitions.metrics.length; i++) {
+        const inputMetric = definitions.metrics[i]
+        const sortedName = sortedNames[i]
+
+        if (inputMetric.name != sortedName) {
+            validations.push(`Telemetry Metrics are not sorted. Expected: ${sortedName}, Found: ${inputMetric.name}`)
+            break
+        }
+    }
```

Comment:
I might be being too picky here but should we also sort `types` used in `metrics`. e.g. codewishperer metrics have so many types.. 
https://github.com/aws/aws-toolkit-common/blob/c07e6a5a64381fcfaa89ef5130fe27451137ccfd/telemetry/definitions/commonDefinitions.json#L2969

Reply:
Yup, both `types` and `metrics` get sorted. There are some tests that exercise each of these.

Reply:
I might have missed the logic but i meant `types` defined in `metadata` of `metrics`

Reply:
Oh I see. Right, I only sorted the root `types` field. 

Yes, that is worthwhile for the same reasons. Let me take a look into that.

Reply:
Done